### ISSUE TITLE
Dynamically generated 'uri' property

### ISSUE THREAD

**Issue by otsakir** [RestComm/Restcomm-Connect#1976]
Why store 'uri' property in the database and not generate it on the fly? It will much more flexible, won't really affect performance and reduce dependencies on data. 

A central mechanism could be used, customized behavior in the converters, removal from DAO layer etc.

Thoughts ?

**Comment by gvagenas** [comment]
@otsakir I agree with your point but the dependency to data layer cannot be completely removed, we still need to check the CDR of a call given the CallSID for example.

Even though I agree, its not clear to me the benefits of that.

PR #1801 is pending on that now?

**Comment by gvagenas** [comment]
Also, regarding the implementation @otsakir , I am not sure if uri dynamic generation can be done with a central mechanism or not. That really depends on the details, the converters etc and its better to check this during implementation.
I believe though that a central mechanism would be much better.

**Comment by abdulazizali77** [comment]
@otsakir, As it were a lot of the uris seem to be rewritten on the fly anyways but seems too adhoc. Inline with #898 and or discussions about header based EndPoints, I agree with @gvagenas that we should need to look at the specific implementation details/dependencies/requirements of each EndPoint and abstract out from that IF we want to have a central uri rewriting mechanism for all, and potentially memoize the uri calls too.

**Comment by otsakir** [comment]
@abdulazizali77 , definitely this will be implemented in a per-case fashion. I mentioned a central mechanism not as a black box that will decide arbitrarily which 'uri' property each entity will get but as a place where some common properties would go for example the '/restcomm' or the '/2012-04-24' parts, the addition of '.json' suffix or other aspects of 'url' generation that we can't imagine as we speak. 

Also, i can't really tell whether the 'uri' should be determined at the converters or the DAO layer or whether for all entities the same practice is used. However, it makes sense to me not to store it in the DB.

@gvagenas, it's not clear to me what was decided regarding #1801.

**Comment by otsakir** [comment]
After discussions with @gvagenas and @abdulazizali77 decided to merge #1801 and proceed with #1976 when time permits as a separate issue. Agreed that the 'uri' should be removed from DB for most entities. The implementation details are not clear and may involve a central mechanism as well as other per-entity modifications that will be decided by the undertaker of the task. 

Possibly, #898 should be considered too in the same epic.
